stacks-signer/src/runloop.rs

```diff
@@ -90,6 +105,10 @@ pub struct RunLoop<C> {
     pub state: State,
     /// Wether mainnet or not
     pub mainnet: bool,
+    /// Observed blocks that we have seen so far
+    pub blocks: HashMap<Vec<u8>, BlockInfo>,
```

Comment:
I think this `Vec<u8>` is a `Sha512Trunc256Sum`?

Reply:
Also, does `blocks` ever get garbage-collected, or does it grow forever?

Reply:
I have a remove block on each sign completion. However, if we never complete a sign, it will grow indefinitely. I have left this as is because the state ticket is on Aaron's todo list and didn't want to get into it yet.

Reply:
And you are right it is a Sha512Trunc256Sum. I know I had it as this before, but it broke when I did this so I reverted it. I will try again, but it has to do with the serialization/deserialization or something of the message to sign.

Reply:
Done :D